File: security-micro/src/main/java/pl/damiankaplon/service/SecurityService.java
package pl.damiankaplon.service;


import io.smallrye.jwt.auth.principal.ParseException;
import org.bson.types.ObjectId;

import javax.security.auth.login.LoginException;
import java.util.Optional;

public interface SecurityService {

    record Credentials(String login, String password){}
    record BearerToken(String token){}
    record BearerTokenPair(BearerToken token, BearerToken refreshToken){}
    record Registration(String email, String password, String name, String surname){}
    record AccountDTO(String email, String name, String surname){}

    BearerTokenPair login(Credentials credentials) throws LoginException;
    ObjectId register(Registration dto) throws LoginException;
    BearerToken refreshToken(BearerToken token, BearerToken refreshToken) throws ParseException, LoginException;
    Optional<AccountDTO> getAccountById(String id);

}

File: security-micro/src/main/kotlin/pl/damiankaplon/service/SecurityService.kt
package pl.damiankaplon.service

import java.util.UUID

data class Credentials(val login: String, val password: String)
data class BearerToken(val token: String)
data class AccountDTO(val uuid: UUID, val email: String, val name: String?, val surname: String?, val password: String)
interface SecurityService {
    fun getBearerToken(credentials: Credentials): BearerToken
    fun registerAccount(dto: AccountDTO): UUID
}